\documentclass[UTF8]{beamer}
\usepackage{graphicx, color}
\usepackage{algorithm2e}
\usepackage{zhspacing}
\usepackage{amsmath}

\usepackage{underscore}
\usetheme{JuanLesPins}
\usepackage{fontspec}
\setsansfont{Microsoft YaHei}

\usepackage{enumerate}

\AtBeginSection[]{
  \frame{
    \frametitle{Next}
    \tableofcontents[currentsection, subsectionstyle=show/shaded/hide]
  }
}

\AtBeginSubsection[]{
  \frame{
    \frametitle{Next}
    \tableofcontents[currentsubsection]
  }
}

\title{Python in Bioinformatics}

\author{Gang Chen\\ chengang@bgitechsolutions.com}

\logo{\includegraphics[width=1.3cm]{bgi-logo.png}\includegraphics[width=2.5cm]{cuhklogo.png}}
\date{\today}




\begin{document}

\begin{frame}
\titlepage
\end{frame}

\begin{frame}[t]\frametitle{Outline}
\tableofcontents[hideallsubsections]
\end{frame}

\section{Python Package}
\begin{frame}
  \frametitle{Python Package}
  \begin{itemize}
    \item python setup
    \item easy_install
    \item pip
  \end{itemize}
\end{frame}

\begin{frame}
  \frametitle{Python Package Development}
  \begin{itemize}
    \item Tutorial:https://packaging.python.org/en/latest/distributing.html
    \item Example:https://github.com/pypa/sampleproject
  \end{itemize}
\end{frame}

\begin{frame}
  \frametitle{Example}
  see hello directory
\end{frame}

\section{Python based Bioinformatics Projects}
\subsection{Scipy}
\begin{frame}
  \frametitle{scipy project}
\end{frame}

\begin{frame}
  \frametitle{Installation and Example}
\end{frame}
\subsection{BioPython}
\begin{frame}
  \frametitle{BioPython}
  Biopython is a set of freely available tools for biological computation
  written in Python by an international team of developers.
\end{frame}

\begin{frame}
  \frametitle{Installation and Example}
  \begin{itemize}
    \item Download
    \item python setup.py
    \item see bio.py as an example
  \end{itemize}
\end{frame}


\subsection{Machine Learning}
\begin{frame}
  \frametitle{Python for Machine Learning}
  \begin{itemize}
    \item scikit-learn
    \item pyml
  \end{itemize}
\end{frame}

\begin{frame}
  \frametitle{PyML}
  PyML is an interactive object oriented framework for machine learning written
   in Python. PyML focuses on SVMs and other kernel methods.
\end{frame}

\begin{frame}
  \frametitle{scikit-learn}
  \begin{itemize}
    \item Simple and efficient tools for data mining and data analysis
    \item Accessible to everybody, and reusable in various contexts
    \item Built on NumPy, SciPy, and matplotlib
    \item Open source, commercially usable - BSD license
  \end{itemize}
\end{frame}

\subsection{Network Visualization and Analysis}
\begin{frame}
  \frametitle{igraph for network visualization}
  \begin{block}{igraph}
    igraph is a collection of network analysis tools with the emphasis on
    efficiency, portability and ease of use. igraph is open source and free.
    igraph can be programmed in GNU R, Python and C/C++.
  \end{block}

  igraph is implemented in C++, but can be programmed in R, Python and C/C++.
\end{frame}

\begin{frame}
  \frametitle{Install and Example}
  \begin{itemize}
    \item pip install python-igraph
    \item Tutorial: http://igraph.org/python/doc/tutorial/tutorial.html
  \end{itemize}
\end{frame}

\section{Bioinformatics in the Cloud using Python}

\subsection{Clouding Computing and Bioinformatics}

\begin{frame}{Clouding Computing}
  \begin{itemize}
    \item Amazon Web Service: aws.amazon.com
    \item Aliyun: aliyun.com
    \item Google Compute Engine: cloud.google.com
    \item Microsoft Azure: azure.microsoft.com
    \item \ldots
  \end{itemize}
\end{frame}

\begin{frame}
  \frametitle{NCI Cancer Genomics Cloud Pilots}
  \begin{block}{Current Needs in Cancer Research}
    The challenges posed by the need to disseminate, manage, and interpret
    large, multi-scale data pervade efforts to advance understanding of cancer
    biology and apply that knowledge in the clinic.
  \end{block}
\end{frame}

\begin{frame}{Bioinformatics in the Cloud}
  \begin{itemize}
    \item DNANexus: DNANexus.com
    \item SBGenomics: SBGenomics.com\\
      rabix: rabix.org
    \item GeneDock: GeneDock.com
    \item L3-Bioinformatics: l3-bioinfo.com
    \item tute genomics, Variant Analysis from Qiagen, \ldots
  \end{itemize}
\end{frame}

\subsection{Python SDK of DNANexus}

\begin{frame}
  \frametitle{Overview}
  https://wiki.dnanexus.com/Developer-Portal
\end{frame}

\begin{frame}
  \frametitle{Installation}
  pip install dxpy
\end{frame}

\subsection{Rabix from SBGenomics}

\begin{frame}
  \frametitle{Reproducible Research}
  \begin{block}{Reproducible Research}
The goal of reproducible research is to tie specific instructions to data
analysis and experimental data so that scholarship can be recreated, better understood and verified.
  \end{block}
  \begin{block}{References}
    \begin{itemize}
      \item https://www.coursera.org/course/repdata
      \item http://cran.r-project.org/web/views/ReproducibleResearch.html
    \end{itemize}
  \end{block}
\end{frame}

\begin{frame}
  \frametitle{Rabix Project}
  see rabix_ismb.pdf
\end{frame}

\end{document}
