\documentclass[UTF8]{beamer}
\usepackage{graphicx, color}
\usepackage{algorithm2e}
\usepackage{zhspacing}
\usepackage{amsmath}

\usepackage{underscore}
\usetheme{JuanLesPins}
\usepackage{fontspec}
\setsansfont{Microsoft YaHei}

\usepackage{enumerate}

\AtBeginSection[]{
  \frame{
    \frametitle{Next}
    \tableofcontents[currentsection, subsectionstyle=show/shaded/hide]
  }
}

\AtBeginSubsection[]{
  \frame{
    \frametitle{Next}
    \tableofcontents[currentsubsection]
  }
}

\title{Perl in Bioinformatics}

\author{Gang Chen\\ chengang@bgitechsolutions.com}

\logo{\includegraphics[width=1.3cm]{bgi-logo.png}\includegraphics[width=2.5cm]{cuhklogo.png}}
\date{\today}




\begin{document}

\begin{frame}
\begin{block}{Assignment of Perl}
\begin{itemize}
\item Write a program to print Fibonacci sequence.The length of output sequence
is specified by the first command line parameter. (fibonacci.pl)
\item Try to optimize the pos_annotate.pl as much as you can. (pos_annotateV3.pl)
\end{itemize}
All assignments should be sent to chengang@genomics.cn
\end{block}
\end{frame}

\begin{frame}
\titlepage
\end{frame}
\begin{frame}[t]\frametitle{Outline}
\tableofcontents[hideallsubsections]
\end{frame}


\section{Examples}

\subsection{System Administration}

\begin{frame}[fragile]{Command Options of perl}
\begin{block}{Options}
  \begin{itemize}
    \item -e
    \item -n
    \begin{verbatim}
while (<>) {
  # your code goes here
}
    \end{verbatim}
    \item -p
\begin{verbatim}
  while (<>) {
      # your code goes here
    } continue {
      print or die "-p destination: $!\n";
    }
\end{verbatim}
  \end{itemize}
\end{block}
\end{frame}
%--- Next Frame ---%

\begin{frame}[fragile]{Process file content}
\begin{block}{Adding Line Number to file content}
  \begin{verbatim}
    perl -ne 'print "$. $_"' names.txt
    perl -pe '$_ = "$. $_"' names.txt
  \end{verbatim}
\end{block}
\end{frame}
%--- Next Frame ---%

\subsection{CGI Programming}
\begin{frame}[t]{CGI Programming}
    \begin{block}{CGI}
        Common Gateway Interface (CGI) is a standard environment for web servers to interface with executable programs installed on a server that generate web pages dynamically.
    \end{block}

\end{frame}
%--- Next Frame ---%


\section{Perl Modules}

\subsection{Usage of Perl Modules}

\begin{frame}
  \frametitle{Web Scraping in Perl}
  \begin{block}{LWP}
    The libwww-perl collection is a set of Perl modules which provides a simple
    and consistent application programming interface (API) to the World-Wide Web.

    https://metacpan.org/pod/LWP
  \end{block}
\end{frame}

\begin{frame}
  \frametitle{Example: lwp.pl}
\begin{block}{Files}
  \begin{itemize}
    \item pubmedids.txt: A list of pubmed ids
    \item lwp.pl: Get publication titles of these ids from PubMed.
  \end{itemize}
\end{block}
\end{frame}

\subsection{Management of Perl Modules}

\begin{frame}
  \frametitle{Searching}
  \centerline{www.CPAN.org}
\end{frame}

\begin{frame}
  \frametitle{Installation}
\begin{columns}
  \begin{column}{.5\textwidth}
    \begin{itemize}
      \item cpan command: Linux, Mac OS and Windows(Strawberry)
      \item perl -MCPAN -e shell
      \item cpanm
      \item from source
    \end{itemize}
  \end{column}
  \begin{column}{.5\textwidth}
    \begin{itemize}
      \item ppm: ActivePerl
    \end{itemize}
  \end{column}
\end{columns}
\end{frame}



\subsection{Development of Perl Modules}

\begin{frame}
  \frametitle{Development}
  see PUBMED.pm and run.pl
\end{frame}

\subsection{Submit your modules to CPAN}

\begin{frame}
  \frametitle{Why?}
  \begin{itemize}
    \item Easy the installation of your module
    \item Share your work with the community
    \item Increase your importance in the community
    \item Get response form the community to improve your module
  \end{itemize}
\end{frame}

\begin{frame}
  \frametitle{How?}
  http://www.cpan.org/modules/04pause.html
\end{frame}

\section{SNP Annotation}

\begin{frame}
  \frametitle{SNP Annotation}
  \begin{block}{Task}
    Given a list of genome positions, add corrsponding gene symbol to each
    position.
  \end{block}
  \begin{block}{Files}
    \begin{itemize}
      \item Example input file: pos.txt
      \item Annotation database: refGene.txt
      \item Example script: pos_annotate.pl
      \item Optimized Script: pos_annotateV2.pl
    \end{itemize}
  \end{block}
\end{frame}

\subsection{A Perl Script for SNP Annotation}
\begin{frame}
  \frametitle{pos_annotate.pl}
\end{frame}

\begin{frame}
  \frametitle{pos_annotateV2.pl}
\end{frame}

\subsection{SNPAnno module}

\begin{frame}
  \frametitle{SNPAnno module}
  \begin{itemize}
    \item SNPAnno.pm
    \item run_SNPAnno.pl
  \end{itemize}
\end{frame}

\section{Bioinformatics Project in Perl}
\subsection{Annovar}
\begin{frame}
  \frametitle{Annovar}
  \begin{block}{Annovar}
    ANNOVAR: Functional annotation of genetic variants from high-throughput
    sequencing data
  \end{block}
  \centerline{http://www.openbioinformatics.org/annovar/}
\end{frame}

\subsection{Cirocs}
\begin{frame}
  \frametitle{Circos}
  \begin{block}{Circos}
    Circos is a software package for visualizing data and information. It
    visualizes data in a circular layout â€” this makes Circos ideal for
    exploring relationships between objects or positions.
  \end{block}
  \centerline{http://circos.ca}
\end{frame}

\begin{frame}
  \includegraphics[width=\textwidth]{circos.jpg}
\end{frame}

\subsection{BioPerl}

\begin{frame}
  \frametitle{BioPerl}
  \centerline{http://bioperl.org}
\end{frame}

\begin{frame}
  \frametitle{Installation}
  \begin{itemize}
    \item from source
    \item cpan (recommended)
  \end{itemize}
\end{frame}


\begin{frame}
  \centerline{\huge{Thanks!}}
\end{frame}
\end{document}
